dashboard/app/controllers/rubrics_controller.rb

```diff
@@ -68,6 +68,16 @@ def submit_evaluations
       end
     end
 
+    # Update the main TeacherFeedback
+    teacher_feedback = TeacherFeedback.find_or_create_by!(student_id: permitted_params[:student_id], teacher_id: current_user.id, script: rubric.get_script_level.script, level: rubric.level)
+
+    if permitted_params[:keep_working]
+      teacher_feedback.update(review_state: TeacherFeedback::REVIEW_STATES.keepWorking)
+    else
+      teacher_feedback.update(review_state: nil)
+    end
+    teacher_feedback.save!
```

Comment:
I think I am missing something here. why is there new backend functionality being added when we submit evaluations, was this not getting saved prior to this PR? the PR description and Jira item seem to mostly cover UI changes so I am not seeing the connection here.

Reply:
The 'Keep Working' checkbox requires that the student be marked as such. Progress is marked via teacher feedback. So, to support the keep working checkbox, it required the ability to post progress state relative to a rubric. Happy to remove the checkbox and this solution and move it to another PR.

Reply:
Another way to go about it would be to use the existing feedback api to set 'keep working' which would mean posting to both our submit and the teacher feedbacks api endpoint and then querying both on load. I'd need to pass down the script and level ids to craft the requests instead of using the rubric to find them. Could add them to the rubric info and make two requests and forgo the new routes.

Reply:
This is the existing route: https://github.com/code-dot-org/code-dot-org/blob/b728fdb10e2134fb8dbeb3568668bda2550054be/dashboard/app/controllers/api/v1/teacher_feedbacks_controller.rb#L56-L69

So, an initial call to this to mark the progress state before submitting the rubric feedback would accomplish this just as well after updating it to pass down the level/script id.

Reply:
makes sense! I just didn't realize the Keep Working checkbox was new as opposed to being moved from somewhere else in our UI. 

I think it is is dealer's choice on whether to make 1 API call or 2. I kind of like how you have it because the client logic can be a bit simpler, not having to worry about different combinations of request failures. Though I suppose if the existing route got updated in an important way, then our route could get out of date. Always with the tradeoffs 😁